### PR TITLE
Updated hbsd-update-build to support custom kernel configs.

### PR DESCRIPTION
**Pull request by shirkdog**

### CODE REVIEW

File: usr.sbin/hbsd-update/hbsd-update-build [hunk 1]
```diff
@@ -157,17 +158,36 @@ build_source() {
...
-		chroot ${CHROOTDIR} make \
+		if [ ! -z "${KERNCONFIGFILE}" ]; then
+		       	if [ -f "${KERNCONFIGFILE}" ]; then
+				cp -f ${KERNCONFIGFILE} ${CHROOTDIR}/usr/src/sys/$(uname -m)/conf
+				chroot ${CHROOTDIR} make \
+				-C /usr/src \
+				KERNCONF=${kernel} \
+				-sj$(sysctl -n hw.ncpu) \
+				$(echo ${DEVMODE} | \
+					sed 's/yes/-DNO_KERNELCLEAN/') \
```
Comment (lattera) [inline usr.sbin/hbsd-update/hbsd-update-build]: Can you indent the arguments to chroot to follow current style?

Comment (shirkdog) [reply usr.sbin/hbsd-update/hbsd-update-build]: okay

File: usr.sbin/hbsd-update/hbsd-update-build [hunk 2]
```diff
@@ -157,17 +158,36 @@ build_source() {
...
+				debug_print "[-] Could not find kernel config file ${KERNCONFIGFILE}"
+				return 1
+			fi
+		else
+			chroot ${CHROOTDIR} make \
 			-C /usr/src \
 			KERNCONF=${kernel} \
 			-sj$(sysctl -n hw.ncpu) \
 			$(echo ${DEVMODE} | \
 				sed 's/yes/-DNO_KERNELCLEAN/') \
```
Comment (lattera) [inline usr.sbin/hbsd-update/hbsd-update-build]: And here, too, please.

**Comment by opntr** [comment]
@lattera: one you think this patch series is ready, please cherry-pick the changes (instead of merging) (with a squashed commit), it's much cleaner.

**Comment by lattera** [comment]
I'll take a look at this today.

**Pull request closed without merging**